File: sliding-window/MaxConsecutiveOneIII.java
public class MaxConsecutiveOneIII {
    public static int getMaxConsecutiveOneCount(int[] nums,int k){
        int n = nums.length;
        int l =0;
        int r = 0;
        int maxLen = 0;
        int zeroCnt = 0;
        while(r <n){
            if(nums[r] ==0){
                zeroCnt++;
            }
            while(zeroCnt > k){
                if(nums[l] == 0){
                    zeroCnt--;
                }
                l++;
            }
            maxLen = Integer.max(maxLen, r-l+1);
            r++;
        }
        return maxLen;
    }
    public static void main(String[] args) {
        int[] arr = {1,1,1,0,0,0,1,1,1,1,0};
        int k = 2;
        System.out.println(getMaxConsecutiveOneCount(arr, k));
    }
}

File: sliding-window/max_consecutive_one_iii.cpp
#include<iostream>
#include<vector>

int get_max_consecutive_one(std::vector<int>&nums , int k){
    int n = nums.size();
    int left_ptr = 0;
    int right_ptr = 0;
    int max_len = 0;
    int zero_cnt =0;
    while(right_ptr < n){
        if(nums[right_ptr] ==0) zero_cnt++;
        while(zero_cnt>k){
            if(nums[left_ptr] == 0){
                zero_cnt--;
            }
            left_ptr++;
        }
        max_len = std::max(max_len , right_ptr - left_ptr + 1);
        right_ptr++;
    }
    return max_len;
}

int main(){
    std::vector<int>arr = {1,1,1,0,0,0,1,1,1,1,0};
    int k = 2;
    std::cout<<get_max_consecutive_one(arr,k);
    return 0;
}